crates/solver/src/settlement_submission/flashbots_settlement.rs

```diff
@@ -289,16 +291,50 @@ impl<'a> FlashbotsSolutionSubmitter<'a> {
                 gas_estimate,
             );
 
-            match self
-                .flashbots_api
-                .submit_transaction(&raw_signed_transaction)
-                .await
-            {
-                Ok(bundle_id) => {
-                    transactions.push(hash);
-                    previous_tx = Some((gas_price, bundle_id));
+            // submit transaction
+
+            let mut retry = std::cmp::max(SUBMIT_RETRY_ATTEMPTS, 1usize);
+            loop {
```

Comment:
Can this not be a for loop?

Reply:
Actually yes. Refactored to for loop and moved to a separate function.